src/routes/friend.js

```diff
@@ -1,88 +1,164 @@
 const router = require('express').Router();
 const Student = require('../models/student.model');
 const StudentRepository = require('../repository/studentRepository')
+const FriendRequest = require('../models/friendRequest.model');
+const createFriendRequest = require('../factory/friendRequestFactory');
+const FriendRequestRepository = require('../repository/friendRequestRepository');
 
-// fetch the list of friends by email
+/**
+ * @author: Jasmin Guay
+ * Endpoint to send a friend request.
+ * @param {string} senderEmail : email of the sender
+ * @param {string} receiverEmail : email of the students to be added
+ * @return {string} _id  : id of the friend request created.
+ */
+router.route('/add').post(async (req, res) => {
+  const senderEmail = req.body.senderEmail.toString();
+  const receiverEmail = req.body.receiverEmail.toString();
+
+  if (!await StudentRepository.isInFriendList(senderEmail, receiverEmail)) {
+    const newFriendRequest = createFriendRequest({senderEmail, receiverEmail});
```

Comment:
@guaj please check sonarcloud code smells and fix accordingly. This variable is created and never used. Either pass it to the .save(...) method or remove before merging. Thank you!

Reply:
This is actually a bug, this route does not work